wsidicom/instance.py

```diff
@@ -663,65 +782,118 @@ def get_filepointer(
-        of frames in the file. fp needs to be positioned before the BOT.
+    def _read_bot(self) -> Optional[bytes]:
+        """Read basic table offset (BOT). Returns None if BOT is empty
 
         Returns
         ----------
-        list[tuple[int, int]]
-            A list with frame positions and frame lengths. Is empty if BOT is
-            empty.
+        Optional[bytes]
+            BOT in bytes.
         """
-        TAG_BYTES = 4
-        LENGHT_BYTES = 4
         BOT_BYTES = 4
-        if not self._fp.is_little_endian:
-            raise ValueError("fp must be little endian")
-        # Jump to BOT
-        offset_to_value = data_element_offset_to_value(
-            self._fp.is_implicit_VR,
-            'OB'
-        )
-        self._fp.seek(offset_to_value, 1)
-
-        positions: List[Tuple[int, int]] = []
         if self._fp.read_tag() != ItemTag:
             raise WsiDicomFileError(
                 self.filepath,
                 "BOT did not start with an ItemTag"
             )
         bot_length = self._fp.read_UL()
         if bot_length == 0:
-            self._fp.seek(bot_length, 1)
-            return positions
+            return None
         elif bot_length % BOT_BYTES:
             raise WsiDicomFileError(
                 self.filepath,
                 f"BOT length should be a multiple of {BOT_BYTES}"
             )
-
         # Read the BOT into bytes
         bot = self._fp.read(bot_length)
-        bot_end = self._fp.tell()
+        return bot
 
-        # Read through BOT to get offset and length for all but last item
-        this_bot_offset: int = unpack(b"<L", bot[0:BOT_BYTES])[0]
-        for index in range(BOT_BYTES, bot_length, BOT_BYTES):
-            next_bot_offset = unpack(b"<L", bot[index:index+BOT_BYTES])[0]
-            offset = this_bot_offset+TAG_BYTES+LENGHT_BYTES
-            length = next_bot_offset - offset
-            positions.append((bot_end+offset, length))
-            this_bot_offset = next_bot_offset
+    def _read_eot(self) -> bytes:
+        """Read extended table offset (EOT) and EOT lengths.
+
+        Returns
+        ----------
+        bytes
+            EOT in bytes.
+        """
+        EOT_BYTES = 8
+
+        eot_length = self._read_tag_length()
+        if eot_length == 0:
+            raise WsiDicomFileError(
+                self.filepath,
+                "EOT present but empty"
+            )
+        elif eot_length % EOT_BYTES:
+            raise WsiDicomFileError(
+                self.filepath,
+                f"BOT length should be a multiple of {EOT_BYTES}"
+            )
+        # Read the EOT into bytes
+        eot = self._fp.read(eot_length)
+        # Read EOT lengths
+        tag = self._fp.read_tag()
+        if tag != Tag('ExtendedOffsetTableLengths'):
+            raise WsiDicomFileError(
+                self.filepath,
+                f"Expected EOT lengths after EOT, found {tag}"
```

Comment:
Make it a little more googlible: "Expected ExtendedOffsetTableLengths tag after EOT, found {tag}"